### PR TITLE
chore: cover SDK base64 helpers

### LINKED ISSUE #260: Add SDK unit tests for UTF-8 base64 helpers and runtime guards
## Summary

The TypeScript SDK uses dedicated base64 helpers to encode UTF-8 data and decode receipt headers in browser-compatible runtimes. Receipt tests cover common encoded JSON indirectly, but the helpers' Unicode, validation, and missing-runtime branches are not directly covered.

## Current behavior

The helpers:

- encode UTF-8 text through `TextEncoder` and `btoa`;
- decode base64 through `atob`, byte conversion, and `TextDecoder`;
- reject empty, padded incorrectly, whitespace-padded, or non-base64 input;
- return explicit errors when `atob` or `btoa` is unavailable.

## Desired behavior

Add direct, table-driven tests that lock down valid ASCII and Unicode round trips, strict invalid-input handling, and runtime capability errors. Production code should not change unless a very small testability refactor is required.

## Key interfaces

- `encodeUtf8ToBase64()` — ASCII, multibyte Unicode, emoji, and empty-text encoding behavior.
- `decodeBase64ToUtf8()` — valid decoding and strict base64 validation.
- Global `atob` and `btoa` capabilities — deterministic unavailable-runtime errors and restoration after each test.

## Acceptance criteria

- [ ] Tests round-trip plain ASCII, accented text, non-Latin text, and emoji.
- [ ] Tests compare at least one encoded result against a known base64 fixture.
- [ ] Decoding rejects an empty string, leading or trailing whitespace, invalid characters, and invalid padding.
- [ ] Missing `atob` produces the documented decoding-unavailable error.
- [ ] Missing `btoa` produces the documented encoding-unavailable error.
- [ ] Tests restore modified globals even when an assertion fails, so other SDK tests are isolated.
- [ ] Tests use Bun and do not depend on Node `Buffer` for the behavior under test.
- [ ] `cd sdk/typescript && bun run test && bun run typecheck` passes.

## Out of scope

- Changing the receipt schema or cryptographic verification.
- Adding URL-safe base64 support.
- Adding a third-party encoding dependency.
- Changing package publishing or build configuration.

## Blocked by

None — can start immediately.

### PR DESCRIPTION
**Pull request by mittalsonal**
## Summary

- Added direct Bun unit tests for the TypeScript SDK UTF-8 base64 helpers.
- Covered known ASCII fixtures, accented text, non-Latin text, emoji, empty-text encoding, strict invalid input rejection, and missing `atob` / `btoa` runtime errors.
- Restored modified global capabilities after each test to keep the SDK test suite isolated.

Closes #260

## Type Of Change

- [ ] Bug fix
- [ ] Feature
- [ ] Documentation
- [x] Tests
- [ ] Refactor
- [ ] Deployment/config

## Affected Areas

- [ ] Gateway (`gateway/`)
- [ ] Verifier (`verifier/`)
- [ ] Web (`web/`)
- [ ] E2E/tests (`tests/`, `run_e2e.sh`)
- [ ] Benchmarks (`bench/`)
- [ ] Deployment/config (`deploy/`, Docker, env, workflows)
- [ ] Documentation/community files
- [x] TypeScript SDK (`sdk/typescript/`)

## Contributor Checklist

- [x] I kept the change focused and avoided unrelated refactors.
- [x] I updated README/service docs/OpenAPI/env examples when behavior, config, headers, status codes, or public APIs changed. Not applicable; test-only change.
- [x] I did not commit secrets, private keys, funded wallets, API keys, or real production URLs.
- [x] I checked x402/EIP-712 field parity when touching payment context, signatures, timestamps, nonces, chain IDs, receipts, or wallet flow. Not applicable; no protocol behavior changed.
- [x] I checked Docker/Compose/Fly/Vercel docs when touching ports, service names, health checks, or environment variables. Not applicable.

## Verification

List the exact commands you ran and their result.

```text
cd sdk/typescript && npx --yes bun@1.3.14 run test
Result: passed - 31 pass, 1 skip, 0 fail, 85 expect() calls

cd sdk/typescript && npx --yes bun@1.3.14 run typecheck
Result: passed - tsc --noEmit

git diff --cached --check
Result: passed
```

## Screenshots

Not applicable - SDK unit test coverage only, no visible UI changes.

## Notes For Reviewers

Production code is intentionally unchanged. The new tests import the helper module directly and avoid Node `Buffer` for the behavior under test.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive coverage for UTF-8 Base64 encoding and decoding.
  * Added validation for Unicode, empty input, invalid data, and environments without native Base64 support.

> [!NOTE]
> ### Add test coverage for SDK base64 UTF-8 helpers
> Adds a Bun test suite in [base64.test.ts](https://github.com/AnkanMisra/MicroAI-Paygate/pull/409/files#diff-73803b1f1547ad43974865d2324d9d73d06866cb401349e59aee16444af965cf) covering `encodeUtf8ToBase64` and `decodeBase64ToUtf8`. Tests include round-trips for ASCII, accented Latin, Devanagari, and emoji strings, known-good decode fixtures, empty string handling, invalid input error cases, and errors when `atob`/`btoa` are unavailable (simulated via `Object.defineProperty`/`Reflect.deleteProperty`).
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 5061ffe.</sup>
> 
>

### CODE REVIEW

**Comment by mittalsonal** [comment]
Maintainer note: the CodeRabbit finding about preserving/restoring the original global `atob`/`btoa` property state has been addressed in the latest commit. The branch is mergeable and the code-owned checks are green (`sdk-tests`, `typecheck` via the workflow, dependency review, and CodeQL). Happy to make any further adjustments if you spot anything during review.

**Comment by AnkanMisra** [comment]
@codex review

**Review by AnkanMisra — CHANGES_REQUESTED** [review]
https://github.com/AnkanMisra/MicroAI-Paygate/pull/409#pullrequestreview-4896648071

File: sdk/typescript/src/__tests__/base64.test.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import { afterEach, describe, expect, it } from "bun:test";
```
[the remark this answers is not in the document]
Comment (AnkanMisra) [reply sdk/typescript/src/__tests__/base64.test.ts:1]: Both retained commits now include the required co-author trailer.

[the remark this answers is not in the document]
Comment (AnkanMisra) [reply sdk/typescript/src/__tests__/base64.test.ts:1]: Rewrote both retained commit subjects with the allowed `chore:` prefix.

File: sdk/typescript/src/__tests__/base64.test.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    expect(decodeBase64ToUtf8("SGVsbG8sIHdvcmxkIQ==")).toBe("Hello, world!");
+    expect(decodeBase64ToUtf8("4pyF")).toBe("\u2705");
+  });
+
+  it("encodes empty text to an empty base64 payload", () => {
+    expect(encodeUtf8ToBase64("")).toBe("");
+  });
+
+  it("rejects empty, whitespace-padded, invalid-character, and invalid-padding input", () => {
+    for (const value of ["", " aGVsbG8=", "aGVsbG8= ", "abc@", "ab=", "abc==", "a==="]) {
```
[the remark this answers is not in the document]
Comment (AnkanMisra) [reply sdk/typescript/src/__tests__/base64.test.ts]: Reverted this behavior change to match issue #260: empty decode input remains invalid, and the PR is test-only again.

**Review by AnkanMisra — APPROVED** [review]
Rechecked at 5061ffe. The PR is test-only again, matches issue #260, and the SDK tests and typecheck pass. The remaining Vercel status is fork authorization rather than an SDK code failure.

**Review by AnkanMisra — APPROVED** [review]
